src/main/java/strategy/handlers/SupplyHandler.java

```diff
@@ -0,0 +1,20 @@
+package strategy.handlers;
+
+import db.Storage;
+import model.FruitTransaction;
+import strategy.OperationHandler;
+
+public class SupplyHandler implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction transaction) {
+        if (transaction == null) {
+            throw new IllegalArgumentException("Transaction cannot be null");
+        }
+
+        if (transaction.getQuantity() <= 0) {
+            throw new IllegalArgumentException("Quantity must be a positive number");
+        }

```

Comment:
these checks would make sense in each handler, let's create a utility method for this somewhere, maybe in operation handler, and reuse it across handlers